lib/internal/file.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+const {
+  DateNow,
+  NumberIsNaN,
+  ObjectDefineProperties,
+  ObjectPrototypeHasOwnProperty,
+  Symbol,
+  SymbolToStringTag,
+} = primordials;
+
+const {
+  Blob,
+} = require('internal/blob');
+
+const {
+  customInspectSymbol: kInspect,
+  emitExperimentalWarning,
+  kEnumerableProperty,
+  kEmptyObject,
+  toUSVString,
+} = require('internal/util');
+
+const {
+  codes: {
+    ERR_INVALID_THIS,
+    ERR_MISSING_ARGS,
+  },
+} = require('internal/errors');
+
+const {
+  inspect,
+} = require('internal/util/inspect');
+
+const kState = Symbol('kState');
+
+function isFile(object) {
+  return object?.[kState] !== undefined || object instanceof Blob;
+}
+
+class File extends Blob {
+  constructor(fileBits, fileName, options = kEmptyObject) {
+    emitExperimentalWarning('buffer.File');
+
+    if (arguments.length < 2) {
+      throw new ERR_MISSING_ARGS('fileBits', 'fileName');
+    }
+
+    super(fileBits, options);
+
+    let lastModified;
+
+    if (ObjectPrototypeHasOwnProperty(options, 'lastModified')) {
```

Comment:
Browsers doesn't seem to care if it's a own or inherited property, only if its value is `undefined`: 
```js
function MyClass(){};
MyClass.prototype.lastModified=2;
console.log(new File([], null, new MyClass).lastModified) // 2
```

Tried with using a getter, it should only be called once:

```js
let counter = 0;
console.log(new File([], null, { get lastModified() {counter++; return 3} }).lastModified) // 3
console.log(counter); // 1
```

```suggestion
    let { lastModified } = options;

    if (lastModified !== undefined) {
```